RecoTracker/FinalTrackSelectors/plugins/TrackerTrackHitFilter.cc

```diff
@@ -309,7 +315,7 @@ namespace reco {
           pxlTPLProbXYQ_(iConfig.getParameter<double>("PxlTemplateProbXYChargeCut")),
           pxlTPLqBin_(iConfig.getParameter<std::vector<int32_t> >("PxlTemplateqBinCut")),
           PXLcorrClusChargeCut_(iConfig.getParameter<double>("PxlCorrClusterChargeCut")),
-          siStripClusterInfo_(consumesCollector()),
+          siStripClusterInfo_(isPhase2_ ? SiStripClusterInfo() : consumesCollector()),
```

Comment:
if there's a better way of avoiding the `esConsumes` call here, please let me know.

Reply:
maybe @makortel can chime in here, I dont see any better solution

Reply:
Looks reasonable. A possible variation would be to use `std::unique_ptr` or `std::optional` for `siStripClusterInfo_`, that would allow avoiding the construction of the object altogether (and the addition of the default constructor). I leave it to your consideration which approach would seem the best.

Reply:
thanks for the suggestion! `std::optional` offers for an elegant solution that I was not aware of. I applied it in the last push.